mathesar_ui/src/sections/table-view/Header.svelte

```diff
@@ -88,16 +126,45 @@
       resetPositions: oldSize === 0 || group.size === 0,
     });
   }
+
+  let newColumnDropdownIsOpen = false;
+  let newColumnName = '';
+  function addColumn() {
+    const newColumn:TableColumn = {
+      name: newColumnName,
+      type: 'varchar',
+      index: columns.data.length,
+      nullable: true,
+      primaryKey: false,
+      validTargetTypes: null,
+    };
+    dispatch('addColumn', newColumn);
+    newColumnDropdownIsOpen = false;
+    newColumnName = '';
+  }
+
+  function isColumnPresent(name:string) {
+    return columns.data?.some((col) => col.name.toLowerCase() === name?.toLowerCase());
+  }
+  $: isDuplicateColumn = isColumnPresent(newColumnName);
 </script>
 
-<div bind:this={headerRef} class="header">
+<Resizeable
+  bind:this={resizer}
+  target={resizerRef}
+  directions={['e']}
+/>
+<div id="columns-header" bind:this={headerRef} class="header">
   <div class="cell row-control" style="width:{DEFAULT_COUNT_COL_WIDTH + paddingLeft}px;">
   </div>
 
-  {#each columns.data as column (column.name)}
-    <div class="cell" style="
-      width:{columnPosition.get(column.name).width}px;
-      left:{columnPosition.get(column.name).left + paddingLeft}px;">
+  {#each columns.data as column, i (column.name)}
+    <div
+      bind:this={resizerRefs[i]}
+      on:mouseenter={(e) => showResizerForColumn(e, column)}
```

Comment:
This changes the resizer instance from column to column.
Currently, this logic leads to errors and unexpected behaviour while trying to resize columns.

<a href="https://www.loom.com/share/b9955bbc1c3c4fca8bb11aad742661cd"> <p>Issues with resizing columns - Watch Video</p> <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/b9955bbc1c3c4fca8bb11aad742661cd-with-play.gif"> </a>

![Screenshot 2021-08-17 at 12 58 04 AM](https://user-images.githubusercontent.com/11850603/129618902-9a100b08-c7fc-4fbf-80be-aa48cda9857e.png)


Once we have an action for the movable package, it might be better to:
  * create a separate column cell component, and handle the resizing logic within it.
  * save the column size while resizing instead of waiting until resize completes
  * create the resize instance on column component mount instead of during mouseover, and destroy the resize instance during component destroy

Reply:
Good afternoon. I have addressed the requested changes as follows:

- [x] create a separate column cell component, and handle the resizing logic within it.
(https://github.com/pegnott/mathesar/blob/307418dcc4344b3c9b7d937cc21a0b84e2f360c2/mathesar_ui/src/sections/table-view/HeaderCell.svelte)

- [x] save the column size while resizing instead of waiting until resize completes (https://github.com/pegnott/mathesar/blob/307418dcc4344b3c9b7d937cc21a0b84e2f360c2/mathesar_ui/src/components/common/actions/moveable.ts#L44-L47)

- [x] create the resize instance on column component mount instead of during mouseover (https://github.com/pegnott/mathesar/blob/307418dcc4344b3c9b7d937cc21a0b84e2f360c2/mathesar_ui/src/components/common/actions/moveable.ts#L42) 

- [x] destroy the resize instance during component destroy (https://github.com/pegnott/mathesar/blob/307418dcc4344b3c9b7d937cc21a0b84e2f360c2/mathesar_ui/src/components/common/actions/moveable.ts#L35)